File: modules/production/exercises/02_cache/solution/cpp/main.cpp
#include <iostream>
#include <map>
#include <string>

int main() {
    std::map<std::string, std::string> cache;
    auto get = [&](const std::string& p) {
        if (cache.count(p)) return cache[p];
        cache[p] = "x";
        return std::string("miss");
    };
    get("p");
    get("p");
    std::cout << (cache.count("p") && cache.size() == 1 ? "CACHE_HIT" : "MISS") << "\n";
    return 0;
}

File: modules/production/exercises/02_cache/solution/python/main.py
cache={}
def get(p):
    if p in cache: return cache[p]
    cache[p]="x"; return "miss"
def main():
    get("p"); r=get("p")
    print("CACHE_HIT" if "p" in cache and len(cache)==1 else "MISS")
if __name__=="__main__": main()
